tools/bin/change_dependencies.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:glob/glob.dart';
+import 'package:glob/list_local_fs.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+import 'package:yaml/yaml.dart';
+
+void main(List<String> args) {
+  final root = Platform.script.resolve('../../');
+  final glob = Glob('**pubspec.yaml');
+  final files = glob.listSync(root: root.toFilePath()).whereType<File>();
+  for (final file in files) {
+    final yamlEditor = YamlEditor(file.readAsStringSync());
+    final yaml = yamlEditor.parseAt([]);
+    if (yaml is! YamlMap) {
+      continue;
+    }
+    final dependencies = yaml['dependencies'];
+    if (dependencies is! YamlMap) {
+      continue;
+    }
+    for (final package in dependencies.keys) {
+      if (!packagesToPin.contains(package)) {
+        continue;
+      }
+      yamlEditor.update(
+        ['dependencies', package],
+        {
+          // Some packages contain full test projects that are copied in unit
+          // tests. So, use absolute paths.
+          'path':
+              root.resolve('pkgs/$package/').toFilePath().replaceAll(r'\', '/'),
+        },
+      );
+    }
+    if (yamlEditor.edits.isEmpty) {
+      continue;
+    }
+    yamlEditor.update(['publish_to'], 'none');
+    file.writeAsStringSync(yamlEditor.toString());
+  }
+}
+
+const packagesToPin = {
+  'native_assets_builder',
```

Comment:
@HosseinYousefi Would this be useful for package:jnigen and package:jni as well? So that jnigen is both tested against the published package:jni and against the package:jni in the repo?

Reply:
Usually we release both of the versions together, expecting users to use the latests of both, so I don't think it's necessary.